File: web/src/domain/sources/collectorPlatforms.ts
/**
 * Collector platform ids — FE mirror of `server/domain/collector_platforms.py`.
 * Drift-tested against `COLLECTOR_PLATFORMS` / OpenAPI AccountPlatform.
 */

export const COLLECTOR_PLATFORM_ORDER = [
  "telegram",
  "discord",
  "rss",
  "http",
  "mqtt",
  "email",
] as const;

export type CollectorPlatform = (typeof COLLECTOR_PLATFORM_ORDER)[number];

export function isCollectorPlatform(value: string | null | undefined): value is CollectorPlatform {
  return (
    value != null &&
    (COLLECTOR_PLATFORM_ORDER as readonly string[]).includes(value)
  );
}

File: server/domain/collector_platforms.py
"""Collector platform vocabulary (Input-layer leaf — safe for DDL import).

``ADAPTER_BUILDERS`` keys and SQLite ``PLATFORM_CHECK_VALUES`` must both
come from here so schema and factory cannot drift.
"""

from __future__ import annotations

from typing import Final, Literal

CollectorPlatform = Literal["telegram", "discord", "rss", "http", "mqtt", "email"]

COLLECTOR_PLATFORMS: Final[tuple[CollectorPlatform, ...]] = (
    "telegram",
    "discord",
    "rss",
    "http",
    "mqtt",
    "email",
)

#: Enriched list payloads (bot／feed／mailbox info). Telegram stays serialize-only.
COLLECTOR_PLATFORMS_WITH_LIST_ENRICHMENT: Final[frozenset[str]] = frozenset(
    platform for platform in COLLECTOR_PLATFORMS if platform != "telegram"
)
